packages/jest-snapshot/src/index.js

```diff
@@ -34,16 +34,16 @@ const cleanup = (
 ) => {
   const pattern = '\\.' + EXTENSION + '$';
   const files = hasteFS.matchFiles(pattern);
-  const filesRemoved = files
-    .filter(
-      snapshotFile =>
-        !fileExists(snapshotResolver.resolveTestPath(snapshotFile), hasteFS),
-    )
-    .map(snapshotFile => {
+  let filesRemoved = 0;
+
+  files.forEach(snapshotFile => {
+    if (!fileExists(snapshotResolver.resolveTestPath(snapshotFile), hasteFS)) {
+      filesRemoved++;
       if (update === 'all') {
         fs.unlinkSync(snapshotFile);
       }
-    }).length;
+    }
+  });
```

Comment:
Can you use `reduce` ? To avoid ++ ?

```js
const filesRemoved = files.reduce((acc, snapshotFile) => {
  if (!fileExists(snapshotResolver.resolveTestPath(snapshotFile), hasteFS)) {
    if (update === 'all') {
        fs.unlinkSync(snapshotFile);
    }
    acc.push(snapshotFile)
  }
  return acc;
}, []).length;

Reply:
or

```js
const filesRemoved = files.reduce((acc, snapshotFile) => {
  if (!fileExists(snapshotResolver.resolveTestPath(snapshotFile), hasteFS)) {
    if (update === 'all') {
        fs.unlinkSync(snapshotFile);
    }
    return acc + 1
  }
  return acc;
}, 0);

Reply:
I thought about it. But is it good when we have side effects inside `reducer`?
@SimenB

Reply:
`filesRemoved++;` is technically a side effect as well, I don't think one is necessarily worse than the other in that regard